content-src/components/DiscoveryStreamBase/DiscoveryStreamBase.jsx

```diff
@@ -301,11 +302,8 @@ export class _DiscoveryStreamBase extends React.PureComponent {
     const styles = [];
     return (
       <div className="discovery-stream ds-layout">
-        {layoutRender.map((row, rowIndex) => (
-          <div
-            key={`row-${rowIndex}`}
-            className={`ds-column ds-column-${row.width}`}
-          >
+        {layoutRender.map((row, rowIndex) => {
+          const contents = (
```

Comment:
Changes in this file are wrapping a collection or component in DSDismiss stub.

Reply:
clarifying collection or component , Only possible component allowed here are LIST, GRID, HERO, COLLECTION ,TEXT BANNER and TOP_SITE or there can be more?

Reply:
Any component can be in a collection, and would be something that if the collection is dismissed, it would take everything with it, so for example, if a collection contains a component that's not something that can be sponsored, for the sake of that collection, it's sponsored because it's used along with a collection that is only sponsored.

In the case of a single component that's sponsored, any component that's got a campaign id (is a promo) would become dismiss-able. I don't think this is something we need to enforce on our end.

A component or collection can be dismiss-able via a campaign id (or some other id) it then wraps it in a dsdismiss component. It's up to the server side data to use these components in a way that creates a logical experience to the user.